vita3k/renderer/src/vulkan/scene.cpp

```diff
@@ -251,8 +251,51 @@ static void bind_vertex_streams(VKContext &context, MemState &mem) {
     context.render_cmd.bindVertexBuffers(0, max_stream_idx, buffers, context.vertex_buffer_offsets.data());
 }
 
+#ifdef __APPLE__
+// convert indices for triangle fans to indices for a triangle list
+// needed for metal because it does not support a triangle fan implementation
+template <typename T>
+void triangle_fan_to_triangle_list(void *&indices, size_t &count, const uint32_t instance_count) {
+    // if N is the number of faces, there are N + 2 indices for triangle fans and 3N indices for triangle list
+    if (count < 3)
+        // safety check
+        return;
+
+    uint32_t nb_triangle = count - 2;
+
+    T *old_indices = reinterpret_cast<T *>(indices);
+    indices = new uint8_t[3 * nb_triangle * instance_count * sizeof(T)];
+    T *curr_indices = reinterpret_cast<T *>(indices);
+
+    for (uint32_t curr_instance = 0; curr_instance < instance_count; curr_instance++) {
+        for (uint32_t triangle = 0; triangle < nb_triangle; triangle++) {
+            curr_indices[0] = old_indices[0];
+            curr_indices[1] = old_indices[triangle + 1];
+            curr_indices[2] = old_indices[triangle + 2];
+            curr_indices += 3;
+        }
+        old_indices += nb_triangle + 2;
+    }
+
+    delete[] reinterpret_cast<uint8_t *>(old_indices);
```

Comment:
Fails to free old_indices
```
Vita3K(37840,0x7ff853f38740) malloc: *** error for object 0x600002d78b7c: pointer being freed was not allocated
```
Maybe need to point first element before freeing memory?
`old_indices -= (nb_triangle + 2) * instance_count;`